### ISSUE TITLE
Inconsistent build failures across supported Rust versions

### ISSUE THREAD

**Issue by Anders429** [svartalf/rust-claim#5]
The inclusion of the `edition = "2018"` line in the [`Cargo.toml` manifest](https://github.com/svartalf/rust-claim/blob/3b0ad8a4f7458ea93080fd1deb69d6eb59cdd4af/Cargo.toml#L6) is causing builds to fail on `rustc 1.30.0`. Attempting to use `claim` on this compiler version results in the following error:

```
   Compiling claim v0.4.0                                                                                                                              
error: Edition 2018 is unstable and only available for nightly builds of rustc.                                                                        
                                                                                                                                                       
error: Could not compile `claim`.
```

The solution is to drop the `edition` tag completely. That seems to be best, since it isn't really a 2018 edition crate anyways, if you're targeting and testing against compiler versions from before 2018.

A similar problem is that the crate doesn't build on anything before `rustc 1.6.0`, despite your version compatibility badge claiming support all the way to `1.0.0`. This is because of the use of `#![no_std]`, which wasn't stable before `1.6.0`. This can be resolved using `autocfg`, similar to what I've done in [similar situations](https://github.com/Anders429/str_overlap/blob/c7a47be07a71744c50f9ed7478d5beac598d552a/src/lib.rs#L26). The error message when building on `rustc 1.0.0` supports this:

```
claim-0.4.0/src/lib.rs:1:1: 1:11 error: no_std is experimental
claim-0.4.0/src/lib.rs:1 #![no_std]
                         ^~~~~~~~~~
```

I can put together a PR with my suggested fixes for these issues.

**Comment by svartalf** [comment]
Hi, @Anders429!

That sounds about right, these specific Rust versions are not included in CI and I  guess that's how these issues were missed.

> I can put together a PR with my suggested fixes for these issues.

It would be really helpful, thank you.

**Comment by svartalf** [comment]
Update: I added more rustc versions into CI into 54b97fbc3b233a5d303efe2612f279cf129936cc (already in `master` branch), that should help checking these cases.

**Comment by svartalf** [comment]
Closed by #6
